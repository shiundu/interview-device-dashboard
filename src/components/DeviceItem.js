// @flow

import React from 'react';

type Props = {
    device : {
        name: String,
        value: String,
        unit: String,
        active: boolean,
        timestamp: number
    },
    handleDeviceStatusUpdate: (any) => void
}

export const DeviceItem = function({...props}: Props) {
    let { device, handleDeviceStatusUpdate} = props;
    return (
        <div className='device' style={{backgroundColor: device.active ? 'green' : 'red' }}>
            <p className='deviceTitle'>{device.name.toUpperCase()}</p>
            <span>value: {device.value} {device.unit}</span>
            <span>{device.active}</span>
            <p>{device.timestamp}</p>
            <button 
                className='activeButton'
                onClick={handleDeviceStatusUpdate(device)}
                style={{backgroundColor: device.active ? 'green' : 'red' }}>
                {device.active ? 'Active' : 'Inactive'}
            </button>
        </div>
    )
}
