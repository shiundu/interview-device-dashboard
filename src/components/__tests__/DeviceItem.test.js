// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import {DeviceItem} from '../DeviceItem';

const device = {
    name: '',
    value: '',
    unit: '',
    active: true,
    timestamp: 1010010101
};

const handleDeviceStatusUpdate = () => {}

it('DeviceItem should render correctly', () => {
    const component = renderer.create(<DeviceItem device={device} handleDeviceStatusUpdate={handleDeviceStatusUpdate} />);
  
    expect(component.toJSON()).toMatchSnapshot();
});
