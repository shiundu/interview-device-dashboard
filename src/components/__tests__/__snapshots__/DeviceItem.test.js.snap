// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceItem should render correctly 1`] = `
<div
  className="device"
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <p
    className="deviceTitle"
  >
    
  </p>
  <span>
    value: 
    
     
    
  </span>
  <span />
  <p>
    1010010101
  </p>
  <button
    className="activeButton"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    Active
  </button>
</div>
`;

exports[`should render correctly 1`] = `
<div
  className="device"
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <p
    className="deviceTitle"
  >
    
  </p>
  <span>
    value: 
    
     
    
  </span>
  <span />
  <p>
    1010010101
  </p>
  <button
    className="activeButton"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    Active
  </button>
</div>
`;
